lib/encoding/zstd/stream_cgo.go

```diff
@@ -0,0 +1,46 @@
+//go:build cgo
+
+package zstd
+
+import (
+	"io"
+
+	"github.com/valyala/gozstd"
+)
+
+// Reader is zstd reader
+type Reader struct {
+	d *gozstd.Reader
+}
+
+// NewReader returns zstd reader for the given r.
+func NewReader(r io.Reader) *Reader {
+	return &Reader{
+		d: gozstd.NewReader(r),
+	}
+}
+
+// Close implements io.ReadCloser interface
+func (r *Reader) Close() error {
+	r.d.Release()
```

Comment:
This code is not safe and may cause segmentation fault. zstd reader cannot be re-used after  the `Release` method is called:
https://pkg.go.dev/github.com/valyala/gozstd#Reader.Release

 If it's needed to re-use gozstd.Reader object (with sync.Pool) for example, it's better to call Reset(nil,nil).